src/test/java/com/google/api/codegen/testdata/go/go_library.baseline

```diff
@@ -459,7 +476,14 @@ func (c *Client) MergeShelves(ctx context.Context, req *librarypb.MergeShelvesRe
 
 // CreateBook creates a book.
 func (c *Client) CreateBook(ctx context.Context, req *librarypb.CreateBookRequest, opts ...gax.CallOption) (*librarypb.Book, error) {
-    ctx = insertXGoog(ctx, c.xGoogHeader)
+    vals = []string{}
+    vals = append(vals, fmt.Sprintf("%s=%v", "name", req.GetName()))
+    vals = append(vals, fmt.Sprintf("%s=%v", "book.read", req.GetBook().GetRead()))
+    md := map[string][]string{
+        "x-goog-api-client": c.xGoogHeader,
+        "x-goog-request-params": vals,
+    }
+    ctx = insertMetadata(ctx, md)
```

Comment:
I'm not too excited about creating one more map every call. Do you think we can make `c.xGoogHeader` a `metadata.MD` instead?

Then the calls become
```go
insertMetadata(ctx, c.xGoogHeader)
```
and
```go
reqParams := metadata.Pairs("x-goog-request-params", vals)
insertMetadata(ctx, c.xGoogHeader, reqParams)
```

Reply:
If we're going to change the type of `XGoogHeader`, can we also change it's name? `XGoogHeader` is not quite accurate anymore, if there are multiple `x-goog-*` headers... Maybe `Metadata` instead?